docs/howtos/DataPlane Generation - DPG/09versioning.mdx

```diff
@@ -0,0 +1,536 @@
+import Tabs from "@theme/Tabs";
+import TabItem from "@theme/TabItem";
+
+# Versioning
+
+This doc details what emitters will generate for versioned specs
+
+## Single api Version
+
+If there is just one api version in the spec, we will generate the api surface area for that one version.
+
+<Tabs>
+<TabItem value="typespec" label="TypeSpec" default>
+
+```typespec
+import "@typespec/versioning";
+
+using TypeSpec.Versioning;
+
+@versioned(My.Service.Versions)
+@service
+namespace My.Service;
+
+enum Versions {
+  v2023_11_01: "2023-11-01",
+}
+
+model StableModel {
+  stableFeature: string;
+}
+
+op stableFunctionality(...StableModel): void;
+```
+
+</TabItem>
+
+<TabItem value="python" label="Python">
+
+```python
+import pytest
+from my.service import MyServiceClient, models
+
+client = MyServiceClient(endpoint=..., credential=...)
+# client's api_version will be "2023-11-01"
+
+stable_model = models.StableModel(stable_feature="present")
+print(stable_model)
+client.stable_functionality(stable_model) # call goes through
+
+preview_client = MyServiceClient(endpoint=..., credential=..., api_version="2023-11-01")
+# python allows you to override the api version, even if only one version is defined in the spec
+```
+
+</TabItem>
+<TabItem value="csharp" label="CSharp" >
+
+```csharp
+//ServiceVersion enum
+public enum ServiceVersion
+{
+    /// <summary> Service version "2023-11-01". </summary>
+    V2023_11_01 = 1,
+}
+
+Uri endpoint = new Uri("<https://my-service.azure.com>");
+ServiceClient client = new ServiceClient(endpoint);
+//client's api_version will be "2023-11-01"
+
+StableModel stableModel = new StableModel("<stableFeature>");
+Response response = client.StableFunctionality(stableModel);
+
+Uri endpoint = new Uri("<https://my-service.azure.com>");
+ServiceClientOptions options = new ServiceClientOptions(ServiceVersion.V2023_11_01);
+ServiceClient client = new ServiceClient(endpoint, options);
+//You can specify the service api-version when create client instance. Now client's api_version will be "2023-11-01"
+```
+
+</TabItem>
+<TabItem value="typescript" label="Typescript" >
+
+```typescript
+// there's no apiVersion defined in the all the operations, TypeScript emitter will ignore it.
```

Comment:
All the examples in this document, don't really have an apiVersion parameter in the operation level, typescript emitter will just ignore it. 
@lmazuel @iscai-msft I should if we should add any test cases that apiVersion is actually being defined in the operation level?

Reply:
I'm going to start off without having it in the docs, since people will mainly use templates, and that takes care of it